### PR TITLE
MINOR: Resuming Tasks should not be initialized twice

### PR DESCRIPTION
**Pull request by mjsax**
additionally: removes race condition in StreamThreadTest plus cleanup

### CODE REVIEW

**Comment by mjsax** [comment]
\cc @bbejeck

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java [hunk 1]
```diff
@@ -170,34 +168,34 @@ public void testPartitionAssignmentChangeForSingleGroup() {
...
         thread.start();
         TestUtils.waitForCondition(new TestCondition() {
             @Override
             public boolean conditionMet() {
                 return thread.state() == StreamThread.State.RUNNING;
             }
         }, 10 * 1000, "Thread never started.");
+
         thread.shutdown();
-        assertEquals(thread.state(), StreamThread.State.PENDING_SHUTDOWN);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java:189]: This is a race: `thread.shutdown()` sets state to `PENDING_SHUTDOWN` but thread might migrate to `DEAD` before we do this check.

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java [hunk 2]
```diff
@@ -192,7 +192,6 @@ public void resume() {
             }
             transactionInFlight = true;
         }
-        initTopology();
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java:195]: We `initTopology()` already in `AssignedTasks#maybeResumeSuspendedTask()` L252 within `transitionToRunning()`

**Comment by mjsax** [comment]
Triggered a system test for this: https://jenkins.confluent.io/job/system-test-kafka-branch-builder/1262/

**Comment by guozhangwang** [comment]
```
StreamThreadTest.java:424:65: '{' is not preceded with whitespace. [WhitespaceAround]
```

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java (hunk 2 above)
Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java:195]: Good catch!

**Review by guozhangwang — APPROVED** [review]
LGTM. Please feel free to merge after fixing the checkstyle issue.

**Comment by bbejeck** [comment]
retest this please

**Review by bbejeck — APPROVED** [review]
Thanks @mjsax, LGTM

**Comment by mjsax** [comment]
Added one commit with a minor internal renaming of a method.

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `1.1` branch.

**Pull request merged**